app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ActiveRecord::Base
+  has_many :posts
+  has_many :comments, through: :posts
+  has_many :likes, through: :posts
```

Comment:
- [ ] in the user model the foreign key should be set, this will look something like below:
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'

```